pkg/util/gitutil/git_test.go

```diff
@@ -204,3 +203,49 @@ func createTestRepo(e *ptesting.Environment) {
 	e.RunCommand("git", "branch", "my/content")
 	e.RunCommand("git", "tag", "my")
 }
+
+func TestTryGetVCSInfoFromSSHRemote(t *testing.T) {
```

Comment:
We should rewrite this to be a more Go-idomatic "table driven test".
https://github.com/golang/go/wiki/TableDrivenTests

This is a fancy way to say:

- Declare each test scenario as an instance of an anonymous struct
- Declare all of your test scenarios as a big slice
- Run the tests inside of a for-loop

For example:
```
struct VCSInfo {
    Kind string
    Login string
    Repo string
}
...
func TestTryGetVCSInfo() {
    ...
    struct {
        Remote string
        WantVCSInfo *vcsInfo
    }{
        // SSH Remotes
        { "git@gitlab.com:owner-name/repo-name.git", &VCSInfo{ ... }},
        { "git@bitbucket.org:owner-name/repo-name.git", &VCSInfo{ ... }},

        // HTTPS remotes
        { "https://gitlab-ci-token:dummytoken@gitlab.com/owner-name/repo-name.git", &VCSInfo{ ... }}, 

        // Errors (nil info returned)
        { "asdfasdfasdf", nil },
        { "svn@github.com:owner-name/repo-name.git", nil },
        { "", nil }
    }

    for _, test := range allTests {
        got := TryGetVCSInfo(test.Remote)
        if got != test.Want {
            ...           
        }
    }
```

There are a lot of benefits to this pattern, the most obvious of which is that it's easy to write a lot of tests without any boilerplate.

Reply:
I like this. I'll make this change.